### ISSUE TITLE
WIKI 1 to 6 executed, but get a black screen after uploading, what's wrong, please help

### ISSUE THREAD

**Issue by ksw2404** [chrisweather/RocMQTTdisplay#19]
hi, i have finished wiki points 1 to 6, but till now the display is not working yet, i only get a black screen, i don't know what i did wrong, i just haven't finished wiki point 7 yet, I do not understand that point, because I do not know where to enter this in rocrail, the explanation is also too unclear for that, I have created everything, connected it and uploaded everything and it works, the connections are good, you can see that on the display ie is uploading, but when it's finished you get a black screen, so I forgot something or made a mistake somewhere, can you help me fix the error so that it works and help with wiki point 7, how to do this and where in rocrail.

I hear from you

**Comment by chrisweather** [comment]
Do you see different status messages on the display when you connect the controller to power via USB?
If that's not the case check the wiring of the display.

Did you configure the correct controller board in the Arduino IDE? Which controller do you have?

Here you can find a demo workspace for rocrail where all config is done already for testing the displays: https://github.com/chrisweather/RocMQTTdisplay/tree/main/demo

**Comment by ksw2404** [comment]
Hi,

The cables are connected properly, first there was a startup message when uploading and then the screen went black, but now it stays black immediately. The mqqt broker is also well connected, you can see that messages from rocrail are coming in. So I don't know which step I did wrong, I don't understand step 7 of your wiki, I checked the demo link, but where should you add these files? i will also add the report of the esp8266 data uploader here below, maybe i entered some data incorrectly, maybe i did some steps wrong, i don't know, i did wiki step 1/6 exactly, but maybe something is missing

The controller / board = Node mcu ESP-12.

want to see what was done wrong or what I forgot and explain where to add the demo files in rocrail

The report =


Starting Roc-MQTT-Display...
Loading configuration from 
/rmdcfg.txt

Saving configuration to 
/rmdcfg.txt

Loading sec from 
/rmdsec.txt

Loading template data from 
/rmdtpl.txt

Loading templates from 
/rmdtpl00.txt
/rmdtpl01.txt
/rmdtpl02.txt
/rmdtpl03.txt
/rmdtpl04.txt
/rmdtpl05.txt
/rmdtpl06.txt
/rmdtpl07.txt
/rmdtpl08.txt
/rmdtpl09.txt

Roc-MQTT-Display
Version 1.07

Current MQTT broker IP-adress: 127.0.0.1

FOR CONFIGURATION OPEN: http://RocMQTTdisplayC01
                          or 
                        http://(IP unset)

  Displays enabled: 1 / 8
  Display size: 128 x 32
  Display: 1  connected to multiplexer port (SCx, SDx): 0  Display-ID: D01

OTA Ready -> Port: RocMQTTdisplayC01 at (IP unset)

Webserver started and listening for requests

**Comment by chrisweather** [comment]
Can you access this site in a browser?
FOR CONFIGURATION OPEN: 
http://RocMQTTdisplayC01

Current MQTT broker IP address: 127.0.0.1
This is the local host address.
The controller cannot access this adress from your network.
You must find out the IP address of your PC that is running the MQTT broker and change it in the controller configuration.

The Rocrail demo workspace can be opened with rocrail.

**Comment by ksw2404** [comment]
Hi, the link http://Roc MQTT display C01 works, I added the wifi ip address, but nothing changed, what do I have to change so that the controller can access it? In addition to an ethernet connection, my pc also has a wifi connection, I followed the instruction via youtube when setting up the mqtt broker, it also works if you enter the local ip address, you also see that rocrail sends messages.

I have the following controller by the way: https://www.amazon.nl/gp/product/B01N9RXGHY/ref=ppx_yo_dt_b_asin_title_o01_s01?ie=UTF8&th=1 , it says that it has a wlan and Bluetooth, isn't it the case that the controller has its own ip address, so somewhere there is a setting wrong. But the following things must be entered in the Configuration, are these data correct?

WIFI Devicename
RocMQTTdisplayC01
WIFI reconnect delay (ms)
5000
OTA Hostname
RocMQTTdisplayC01
OTA Port 8266

Do you have to create an OTA file in Arduino ide yourself? and add to the esp8266?.

But if you upload the display you still get the text up to display 1, it does not go any further.

So what do I need to adjust to get the display working?

The demo I don't understand how you add that in rocrail.

Greetings

**Comment by chrisweather** [comment]
You are close to the solution.
I assume rocrail and the MQTT broker run on the same PC. That's why they find each other via localhost or 127.0.0.1
Run a command prompt on your PC and type 'ipconfig / all'
This will tell you the real IP address.
Enter this IP address in the controller configuration.
The display controller needs the IP address of your PC (ofter similar to 192.168.xxx.yyy) in the Configuration to communicate with the broker. 

Extract the rmddemo.zip to a folder on your harddisc. From Rocrail chose File/Open Workspace and select the folder you just created.

**Comment by chrisweather** [comment]
Attached the config for 1 display directly connected to the controller.

 

Von: ksw2404 ***@***.***> 
Gesendet: Montag, 22. August 2022 22:05
An: chrisweather/RocMQTTdisplay ***@***.***>
Betreff: Re: [chrisweather/RocMQTTdisplay] WIKI 1 to 6 executed, but get a black screen after uploading, what's wrong, please help (Issue #19)

 

Hi,

I ran the ip config, many ip addresses come out

Wireless LAN adapter Wi-Fi:

Connection-specific DNS Suffix . : fritz.box
Description . . . . . . . . . . . : Intel(R) Dual Band Wireless-AC 3168
Physical Address. . . . . . . . . : 30-E3-7A-11-D1-A4
DHCP Enabled. . . . . . . . . . . : Yes
Autoconfiguration Enabled . . . . : Yes
Link-local IPv6 Address . . . . . : fe80::5f4:c890:92e2:f93f%4(Preferred)
IPv4 Address. . . . . . . . . . . : 192.168.178.24(Preferred)
Subnet Mask . . . . . . . . . . . : 255.255.255.0
Lease Obtained. . . . . . . . . . : Monday, August 22, 2022 21:09:06
Lease Expires. . . . . . . . . . : Thursday, September 1, 2022 21:17:49
Default Gateway . . . . . . . . . : 192.168.178.1
DHCP Server . . . . . . . . . . . : 192.168.178.1
DHCPv6 IAID . . . . . . . . . . . : 254862202
DHCPv6 Client DUDE. . . . . . . . : 00-01-00-01-20-DE-06-04-C0-7C-D1-FE-26-00
DNS Servers . . . . . . . . . . . : 192.168.178.1

I assume I need the following ip address : IPv4 Address. . . . . . . . . . . : 192.168.178.24(Preferred)

I will enter this and see what will happen

Do I need to change anything in the conforation screen?

Greetings

**Comment by chrisweather** [comment]
Attached the config for 1 display directly connected to the controller.
![image](https://user-images.githubusercontent.com/23246503/186022272-7319f1d6-ad40-4ac6-ac9f-5ae0fcd390dc.jpeg)

**Comment by ksw2404** [comment]
Hi, still nothing, and I copied the report again and no change in IP address.

Starting Roc-MQTT-Display...
Loading configuration from
/rmdcfg.txt

Saving configuration to
/rmdcfg.txt

Loading sec from
/rmdsec.txt

Loading template data from
/rmdtpl.txt

Loading templates from
/rmdtpl00.txt
/rmdtpl01.txt
/rmdtpl02.txt
/rmdtpl03.txt
/rmdtpl04.txt
/rmdtpl05.txt
/rmdtpl06.txt
/rmdtpl07.txt
/rmdtpl08.txt
/rmdtpl09.txt

Roc-MQTT-Display
Version 1.07

Current MQTT broker IP address: 192.168.178.24

FOR CONFIGURATION OPEN: http://RocMQTTdisplayC01
                           or
                         http://(IP unset)

   Display enabled: 1 / 8
   Display size: 128 x 32
   Display: 1 connected to multiplexer port (SCx, SDx): 0 Display ID: D01

OTA Ready -> Port: RocMQTTdisplayC01 at (IP unset)

Web server started and listening for requests

I copied the print screen, but to no avail, it's strange. Also that the controller does not contact the broker

**Comment by chrisweather** [comment]
https://github.com/eclipse/mosquitto/issues/1985

**Comment by ksw2404** [comment]
Hi,

I connected new WeMos D1, a large version, but to no avail.

How is it possible that the Wemos D1 esp8266 does not connect as stated in the report?

here the report

FOR CONFIGURATION OPEN: http://RocMQTTdisplayC01
                          or
                        http://(IP unset)

  Display enabled: 1 / 8
  Display size: 128 x 32
  Display: 1 connected to multiplexer port (SCx, SDx): 0 Display ID: D01

OTA Ready -> Port: RocMQTTdisplayC01 at (IP unset)

As for the Mosquitto broker, I followed all the advice, such as adding a test file, but to no avail. It is a pity that no progress can be made in this way.

But what difference is a raspberry pi going to make, then you still have the localhost ip address hassle? Quite difficult to find a good raspberry pi, there are many types, from €15 to €175.00, there are even external servers that offer Mosquitto broker, that also doesn't work in my opinion

hope you have the solution for me, to solve this problem

Greetings
